.github/workflows/upload-test-seed.yml

```diff
@@ -0,0 +1,94 @@
+name: Upload Test Seed
+
+on:
+  pull_request:
+    paths:
+      - 'studies/**'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      REMOTE_SEED_PATH: 'pull/${{ github.event.pull_request.number }}/seed'
+
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+
+      - name: Comment "Upload In Progress"
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+        with:
+          script: |
+            const commentBody =
+            `## 🔄 Test Seed Upload In Progress
+
+            A new seed file is currently being uploaded for this pull request.
+
+            ### What's Next?
+
+            - The upload process typically takes a few minutes.
+            - Once the upload is complete, this comment will provide the seed details and testing instructions.
+            `
+            const comment = require('.github/workflows/scripts/comment.js')
+            await comment(github, context, commentBody)
+
+      - name: Install dependencies and run tests
```

Comment:
nit: could we split it into?
1. npm install
2. build proto & typecheck
3. tests